syntax = "proto3";

package fixbug;

// 登录请求
message LoginRequest {
    string name = 1;
    string pwd = 2;
}

// 登录响应
message LoginResponse {
    int32 code = 1; // 0成功，-1失败
    string msg = 2;
}

// 定义服务接口
service UserService {
    // 登录方法
    rpc Login(LoginRequest) returns (LoginResponse);
    
    // 获取用户信息
    rpc GetUserInfo(GetUserInfoRequest) returns (GetUserInfoResponse);
}

message GetUserInfoRequest {
    int32 id = 1;
}

message GetUserInfoResponse {
    int32 code = 1;
    string name = 2;
    bool is_vip = 3;
}